python/mxnet/ndarray/numpy/_op.py

```diff
@@ -39,7 +39,7 @@
            'flip', 'around', 'hypot', 'bitwise_xor', 'bitwise_or', 'rad2deg', 'deg2rad', 'unique', 'lcm', 'tril',
            'identity', 'take', 'ldexp', 'vdot', 'inner', 'outer', 'equal', 'not_equal', 'greater', 'less',
            'greater_equal', 'less_equal', 'hsplit', 'rot90', 'einsum', 'true_divide', 'nonzero', 'shares_memory',
-           'may_share_memory', 'diff', 'resize', 'nan_to_num', 'where']
+           'may_share_memory', 'diff', 'resize', 'nan_to_num', 'where', 'diagonal']
```

Comment:
move this to after `full`, same for the function definition and in all other files.

Reply:
seems like you do not need any handlings in Python, so you can declare this as `_np_diagonal` in the backend and you can avoid adding additional Python wrappers.